### DISCUSSION TITLE
Webserver no longer starting

### DISCUSSION THREAD

**Discussion by WraithCadmus** [apache/airflow#24278] [Q&A] [answered]
We had an Airflow install in a lab working well, but it fell over during this weekend. Our DBA says it was using a lot of connections ~96 and it was causing other issues. We killed all processes (Celery Workers, Celery UI, Webserver, and Scheduler), killed any running DB connections, and restarted. Celery seems okay, but the Webserver just hangs, (not tried scheduler). I'm trying to get the thing more verbose by running it from the command line

```
[airflow@airflow-front ~]$ export AIRFLOW_HOME=/var/lib/airflow
[airflow@airflow-front ~]$ airflow webserver -d --port 8080 --pid /var/lib/airflow/airflow-webserver.pid
/usr/local/lib/python3.6/site-packages/sqlalchemy_utils/types/encrypted/encrypted_type.py:18 CryptographyDeprecationWarning: Python 3.6 is no longer supported by the Python core team. Therefore, support for it is deprecated in cryptography and will be removed in a future release.
```

But it just sits there, and I've given it an hour, nothing moves and there's no errors. I initially suspected the DB (a postgres instance), but I can connect to and query the DB from my frontend box. Also commands like `airflow connections list` seem fine, so the application can talk to the DB. I feel I just need some hints as to what it's trying to do, or how to get it more verbose so I can diagnose from there.

It's a pip install on CentOS 7, specifically `pip3 install --ignore-installed "apache-airflow==2.2.5" --constraint "https://raw.githubusercontent.com/apache/airflow/constraints-{{airflow.version}}/constraints-3.6.txt"` I need the `ignore-installed` because of a conflict with a YAML lib, the thing was working though.

**Comment by potiuk** [comment, marked as answer]
Might be that some locks are held if you just abruptly closed some connections. Might require DB server restart. 

And yes. Airflow creates (for now - this is going to change in the future) a lot of DB connections so it is recommended to use pgbouncer to shield Airflow from the Postgres DB: see the note here: https://airflow.apache.org/docs/apache-airflow/stable/howto/set-up-database.html?highlight=database#setting-up-a-postgresql-database 

In both cases you should consult your DB admin (same who complained about the number of connections)

**Reply by WraithCadmus** [reply]
We are using PBBouncer, but that's also in a developmental state, so we're figuring it out as we go.

As for starting the scheduler if memory serves the webserver will run without it, just complain, which is fine right now. I'm trying to isolate one issue at a time.

**Reply by WraithCadmus** [reply]
Dear future people, this was due to a fault in the replication of our Postgres servers behind the bouncer, causing write queries to hang.

**Comment by potiuk** [comment]
BTW. If nothing moves, then it's likely because you have not tried scheduler. Scheduler is the one who "moves things"

**Comment by WraithCadmus** [comment]
Thanks for your help @potiuk . Our DBA says they're having issues with another app, so I'm going to wait until the dust settles on that before even trying this.
